deployers/whiskclient.go

```diff
@@ -20,194 +20,220 @@ package deployers
+		}
+		apiHost.Value = host
+		apiHost.Source = INTERINPUT
+	}
+
+	if len(credential.Value) == 0 && isInteractive == true {
+		cred, err := promptForValue("\nPlease provide an authentication token: ")
+		utils.Check(err)
+		credential.Value = cred
+		credential.Source = INTERINPUT
+
+		// The namespace is always associated with the credential. Both of them should be picked up from the same source.
+		if len(namespace.Value) == 0 || namespace.Value == whisk.DEFAULT_NAMESPACE {
+			ns, err := promptForValue("\nPlease provide a namespace [default value is guest]: ")
+			utils.Check(err)
+			source := INTERINPUT
+
+			if ns == "" {
+				ns = whisk.DEFAULT_NAMESPACE
+				source = DEFAULTVALUE
+			}
+
+			namespace.Value = ns
+			namespace.Source = source
+		}
+	}
+
+	clientConfig = &whisk.Config{
+		AuthToken: credential.Value, //Authtoken
+		Namespace: namespace.Value,  //Namespace
+		Host:      apiHost.Value,
+		Version:   "v1",
+		Insecure:  true, // true if you want to ignore certificate signing
+	}
+
+	if len(credential.Value) == 0 || len(apiHost.Value) == 0 || len(namespace.Value) == 0 {
+		var errStr string
+		if len(credential.Value) == 0 {
+			errStr += wski18n.T("The authentication key is not configured.\n")
+		} else {
+			errStr += wski18n.T("The authenitcation key is set from {{.authsource}}.\n",
+				map[string]interface{}{"authsource": credential.Source})
+		}
+
+		if len(apiHost.Value) == 0 {
+			errStr += wski18n.T("The API host is not configured.\n")
+		} else {
+			errStr += wski18n.T("The API host is {{.apihost}}, from {{.apisource}}.\n",
+				map[string]interface{}{"apihost": apiHost.Value, "apisource": apiHost.Source})
+		}
+
+		if len(namespace.Value) == 0 {
+			errStr += wski18n.T("The namespace is not configured.\n")
+		} else {
+			errStr += wski18n.T("The namespace is {{.namespace}}, from {{.namespacesource}}.\n",
+				map[string]interface{}{"namespace": namespace.Value, "namespacesource": namespace.Source})
+
+		}
+		whisk.Debug(whisk.DbgError, errStr)
+		return clientConfig, utils.NewInvalidWskpropsError(errStr)
+	}
+
+	stdout := wski18n.T("The API host is {{.apihost}}, from {{.apisource}}.\n",
+		map[string]interface{}{"apihost": apiHost.Value, "apisource": apiHost.Source})
+	whisk.Debug(whisk.DbgInfo, stdout)
+
+	stdout = wski18n.T("The auth key is set, from {{.authsource}}.\n",
+		map[string]interface{}{"authsource": credential.Source})
+	whisk.Debug(whisk.DbgInfo, stdout)
+
+	stdout = wski18n.T("The namespace is {{.namespace}}, from {{.namespacesource}}.\n",
+		map[string]interface{}{"namespace": namespace.Value, "namespacesource": namespace.Source})
+	whisk.Debug(whisk.DbgInfo, stdout)
+	return clientConfig, nil
 }
 
-var promptForValue = func (msg string) (string) {
+var promptForValue = func(msg string) (string, error) {
 	reader := bufio.NewReader(os.Stdin)
 	fmt.Print(msg)
 
 	text, _ := reader.ReadString('\n')
 	text = strings.TrimSpace(text)
 
-	return text
+	return text, nil
```

Comment:
Do we anticipate errors from these library calls?  If so, how do we actually catch them here and return a Whisk error?

Reply:
IMO, we do not error no matter what is the input. If there is no input, "" will be the value.